File: BackEnd/src/main/java/com/Checkerly/BackEnd/repository/EventRepository.java
package com.Checkerly.BackEnd.repository;

import java.util.Date;
import java.util.List;
import java.util.Optional;

import com.Checkerly.BackEnd.domain.Organizer;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

import com.Checkerly.BackEnd.domain.Event;

@Repository
public interface EventRepository extends MongoRepository<Event, String> {
    @Query("{ 'assuntoEvento': { $regex: ?0, $options: 'i' } }")
	List<Event> searchAssuntoEvento(String text);
	
	List<Event> findByAssuntoEventoContainingIgnoreCase(String text);
	
	@Query("{ $and: [ { date: {$gte: ?1} }, { date: { $lte: ?2} } , { $or: [ { 'title': { $regex: ?0, $options: 'i' } }, { 'body': { $regex: ?0, $options: 'i' } }, { 'comments.text': { $regex: ?0, $options: 'i' } } ] } ] }")
	List<Event> fullSearch(String text, Date minDate, Date maxDate);

	Optional<Event> findById(String id);
}

File: BackEnd/src/main/kotlin/com/Checkerly/BackEnd/repository/EventRepository.kt
package com.Checkerly.BackEnd.repository

import com.Checkerly.BackEnd.domain.Event
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.mongodb.repository.Query
import org.springframework.stereotype.Repository
import java.util.Date

@Repository
interface EventRepository : MongoRepository<Event?, String?> {
    @Query("{ 'assuntoEvento': { \$regex: ?0, \$options: 'i' } }")
    fun searchAssuntoEvento(text: String?): List<Event?>?

    fun findByAssuntoEventoContainingIgnoreCase(text: String?): List<Event?>?

    @Query("{ \$and: [ { date: {\$gte: ?1} }, { date: { \$lte: ?2} } , { \$or: [ { 'title': { \$regex: ?0, \$options: 'i' } }, { 'body': { \$regex: ?0, \$options: 'i' } }, { 'comments.text': { \$regex: ?0, \$options: 'i' } } ] } ] }")
    fun fullSearch(text: String?, minDate: Date?, maxDate: Date?): List<Event?>?
}
